pkg/controller/common/beat/pod.go

```diff
@@ -0,0 +1,122 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package beat
+
+import (
+	"fmt"
+	"hash"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/container"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/defaults"
+	commonhash "github.com/elastic/cloud-on-k8s/pkg/controller/common/hash"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/volume"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/pointer"
+)
+
+const (
+	CAVolumeName = "es-certs"
+	CAMountPath  = "/mnt/elastic-internal/es-certs/"
+	CAFileName   = "ca.crt"
+
+	ConfigVolumeName = "config"
+	ConfigMountPath  = "/etc/beat.yml"
+	ConfigFileName   = "beat.yml"
+
+	DataVolumeName        = "data"
+	DataMountPathTemplate = "/var/lib/%s/%s/%s-data"
+	DataPathTemplate      = "/usr/share/%s/data"
+
+	// ConfigChecksumLabel is a label used to store beats config checksum.
+	ConfigChecksumLabel = "beat.k8s.elastic.co/config-checksum"
+
+	// VersionLabelName is a label used to track the version of a Beat Pod.
+	VersionLabelName = "beat.k8s.elastic.co/version"
+)
+
+func buildPodTemplate(params DriverParams, defaultImage container.Image, f func(builder *defaults.PodTemplateBuilder), checksum hash.Hash) corev1.PodTemplateSpec {
+	podTemplate := params.GetPodTemplate()
+
+	// Token mounting gets defaulted to false, which prevents from detecting whether user set it.
+	// Instead, checking that here, before the default is applied.
+	if podTemplate.Spec.AutomountServiceAccountToken == nil {
+		t := true
+		podTemplate.Spec.AutomountServiceAccountToken = &t
+	}
+
+	builder := defaults.NewPodTemplateBuilder(podTemplate, params.Type)
+
+	if f != nil {
```

Comment:
Maybe add a comment and explain when this modifier func is nil (in case of the generic beat iiuc)